tests/integration/support/steps.go

```diff
@@ -192,12 +218,22 @@ func (c *ScenarioCtx) InTheResponseShouldHaveEachOneOfThese(argumentType string,
 	return fmt.Errorf("response for %s not equal", argumentType)
 }
 
+// TheNumberOfDispatchedEventsIs represents a step in the feature file
+func (c *ScenarioCtx) TheNumberOfDispatchedEventsIs(count int) error {
+	dispatchedEvents := c.clientWrapper.EventDispatcher.(optlyplugins.EventReceiver).GetEvents()
+	if len(dispatchedEvents) == count {
+		return nil
+	}
+	return fmt.Errorf("dispatchedEvents count not equal")
+}
+
 // ThereAreNoDispatchedEvents represents a step in the feature file
 func (c *ScenarioCtx) ThereAreNoDispatchedEvents() error {
-	if len(c.apiResponse.ListenerCalled) == 0 {
+	dispatchedEvents := c.clientWrapper.EventDispatcher.(optlyplugins.EventReceiver).GetEvents()
+	if len(dispatchedEvents) == 0 {
 		return nil
 	}
-	return fmt.Errorf("listenersCalled should be empty")
+	return fmt.Errorf("dispatchedEvents should be empty")
```

Comment:
nit: Log how many dispatched events we got in addition to saying it should be empty